### PR TITLE
docs: add development faq docs

### PR DESCRIPTION
**Pull request by alessiodionisi**
## Changes

- Added new docs section with development faq

### CODE REVIEW

File: docs/development/development.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  <summary>Answer</summary>
+
+The files used by `furyctl`, in addition to the schema generated in `pkg/apis`, are:
+
+- **Defaults for the distribution**: These are contained in the `defaults` folder. These files define the default values and settings that are used for the configuration, ensuring that the distribution behaves in a predictable manner when no explicit configurations are provided.
+
+- **Templates for the distribution**: These are found in the `templates` folder and include specific templates like Terraform for `EKSCluster`. These templates serve as blueprints for configuring and deploying components (like cloud resources), enabling customizations for different environments.
+
+- **Rules for `furyctl.yaml` changes**: The rules determine which changes are permitted in the `furyctl.yaml` file after initial deployment. These rules are critical for ensuring that modifications to the configuration are safe and consistent with the intended deployment workflow, particularly when managing state changes across environments.
+</details>
```
Comment (nutellinoit) [inline docs/development/development.md:15]: What about `pkg/apis/config`?

File: docs/development/development.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+---
+
+### **How do the rules for immutable fields/migrations work and where are they evaluated in furyctl's code?**
+
+<details>
+  <summary>Answer</summary>
+
+The rules for immutable fields and migrations are evaluated within the core logic of the `furyctl` tool. These rules are defined in the configuration files and enforced by the tool to ensure consistency and prevent misconfigurations.
+
+These rules serve as safety mechanisms during module changes (e.g., switching from the Loki logging system to OpenSearch). Some changes are allowed, while others are not. For example, fields marked as `immutable` will return an error if an attempt is made to change them.
```
Comment (nutellinoit) [inline docs/development/development.md:26]: Where is the state that furyctl checks for immutability?

File: docs/development/development.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+The rules are configured in the `rules` folder, and the commands/scripts executed for each rule are found in `templates/distribution/scripts/pre-apply.sh.tpl`.
+
+Additionally, **Reducers** are special fields rendered inside the template engine that indicate whether a particular feature or module of the distribution has changed. The `.to` and `.from` strings indicate these changes precisely.
```
Comment (nutellinoit) [inline docs/development/development.md:30]: Let's add some info about the key of the value that can be used in the template engine, to know when something changes

File: docs/development/development.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+### **What's inside the Makefile?**
+
+<details>
+  <summary>Answer</summary>
+
+The important commands in the `Makefile` are:
+
+- **`make generate-go-models`**: This command generates Go code from the JSON schema files. The generated code defines the data models used in the codebase, providing a structured representation of the resources and configurations used by `furyctl`. It essentially converts the schema into Go structs, which are essential for interacting with the configuration data programmatically.
+
+- **`make generate-docs`**: This command generates Markdown documentation from the schema files. It extracts the necessary information from the schemas and formats it into human-readable documentation, helping developers and users understand how to configure and use the distribution and resources. This documentation serves as the primary reference for anyone interacting with `furyctl`.
```
Comment (nutellinoit) [inline docs/development/development.md:60]: let's add some infos on how to jumpstart the environment to be able to run these two commands

File: docs/development/development.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+### **What's the purpose of a private schema, and what are the differences from the public schema?**
+
+<details>
+  <summary>Answer</summary>
+
+The `public` schema serves as the base schema, which is shared and visible to all users. It defines the core structure and expected properties for a particular resource or configuration.
+
+The `private` schema is a modified version of the public schema, typically used internally within `furyctl`. It includes additional fields or configurations that should not be exposed in the public configuration files (like `furyctl.yaml`), but are still necessary for certain internal operations or customizations within the codebase.
+
+A notable case where the private schema is used is with the `EKSCluster` resource. Here, a patch is applied to the public schema to add internal fields that are required for `furyctl` to function but are not intended for end-user modification. This separation ensures that sensitive or internal details remain private while maintaining flexibility for internal customization.
```
Comment (nutellinoit) [inline docs/development/development.md:74]: Let's add that the private schema is autogenerated? How it's autogenerated and where are the additional fields taken from?

**Review by nutellinoit — CHANGES_REQUESTED** [review]
Overall LGTM here as well, some minor adjustments 👇

File: docs/development/development.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+---
+
+### **What's inside the Makefile?**
+
+<details>
+  <summary>Answer</summary>
+
+The important commands in the `Makefile` are:
+
+- **`make generate-go-models`**: This command generates Go code from the JSON schema files. The generated code defines the data models used in the codebase, providing a structured representation of the resources and configurations used by `furyctl`. It essentially converts the schema into Go structs, which are essential for interacting with the configuration data programmatically.
```
Comment (ralgozino) [inline docs/development/development.md:58]: we should mention here something about the tool that converts. Where is it? Who maintains it? Known issues?

File: docs/development/development.md (hunk 5 above)
Comment (ralgozino) [reply docs/development/development.md:74]: On top of what @nutellinoit said, I'd add: how could I add a private field to a schema?

File: docs/development/development.md (hunk 6 above)
Comment (alessiodionisi) [reply docs/development/development.md:58]: Documented here: https://github.com/sighupio/go-jsonschema/pull/2

File: docs/development/development.md (hunk 2 above)
Comment (alessiodionisi) [reply docs/development/development.md:28]: I don't think I understand the question. Here, only the rules are written, that part is documented in furyctl.

File: docs/development/development.md (hunk 6 above)
Comment (ralgozino) [reply docs/development/development.md:64]: ok, let's link to the tool repo to make it discoverable, something like "The tool used is an internal development called [go-jsonschema](link)"

**Review by ralgozino — CHANGES_REQUESTED** [review]
There are still unresolved conversations

File: docs/development/development.md (hunk 2 above)
Comment (alessiodionisi) [reply docs/development/development.md:30]: Just added some details about it with @kriive

**Pull request merged**